code/modules/awaymissions/zlevel.dm

```diff
@@ -62,6 +62,13 @@ var/global/list/potentialRandomZlevels = generateMapList(filename = "config/away
 /proc/seedRuins(list/z_levels = null, budget = 0, whitelist = /area/space, list/potentialRuins = space_ruins_templates)
 	if(!z_levels || !z_levels.len)
 		z_levels = list(1)
+
+	for(var/zl in z_levels)
+		var/turf/T = locate(1, 1, zl)
+		if(!T)
+			WARNING("Z level [zl] does not exist - Not generating ruins")
+			return

```

Comment:
A general rule, is if you bother to validate input, you should avoid creating a scenario where what you do with invalid input is different than what the caller would have done if it validated input.

Because to the caller's perspective, if it would have wanted you to just filter it out, it now has to do that, meaning input is checked twice, and one of the checks should be removed.

This is the big balancing act you have to do, if both sides are checking the args, then thats a waste.

In fact, I think this pr is a bad idea in general, ssmapping should be changed to make sure the z-levels exist, or should be changed to make it easier for maps to define what zlevels to seed what on.

If seedRuins is being called with invalid input, that is not a bug in seedRuins, that is a bug in what called it, and the runtime should be allowed to naturally happen rather then waste cpu time checking the same list twice since now somebody will have to go to ssmapping and make it filter this list so that seedRuins doesn't decide to throw out the baby with the bath water because one of the zlevels in a list wasn't valid.